game/state/city/vehiclemission.cpp

```diff
@@ -3067,6 +3013,62 @@ void VehicleMission::takePositionNearPortal(GameState &state, Vehicle &v)
 	                        state, v, v.getPreferredPosition(xPos(state.rng), yPos(state.rng))));
 }
 
+bool VehicleMission::canRecoverVehicle(GameState &state, Vehicle &v, Vehicle &target)
+{
+	// Ground can't recover
+	if (v.type->isGround())
+	{
+		return false;
+	}
+	// Target not crashed or dead or already rescued
+	if (target.isDead() || (!target.crashed && !target.sliding && !target.falling) ||
+	    target.carriedByVehicle)
+	{
+		return false;
+	}
+	// Only X-Com pursues sliding or falling crafts
+	if (!target.crashed && v.owner != state.getPlayer())
+	{
+		return false;
+	}
+	// Can't rescue
+	if (target.owner != state.getAliens() && !v.type->canRescueCrashed)
+	{
+		return false;
+	}
+	// Only X-Com can storm UFO
+	if (target.owner == state.getAliens() && v.owner != state.getPlayer())
+	{
+		return false;
+	}
+	// Find soldier
+	bool needSoldiersButNotFound =
+	    v.owner == state.getPlayer() ||
+	    v.owner->isRelatedTo(target.owner) == Organisation::Relation::Hostile;
+	for (auto &a : v.currentAgents)
```

Comment:
This could be const? (e.g. "const auto &a")

Reply:
Sure (I've only moved this code block from `VehicleMission::start`).